pkg/webhook/pod/util.go

```diff
@@ -138,3 +138,19 @@ func getOwnerStatefulSetForTiDBComponent(pod *core.Pod, kubeCli kubernetes.Inter
 	}
 	return kubeCli.AppsV1().StatefulSets(namespace).Get(ownerStatefulSetName, meta.GetOptions{})
 }
+
+// checkFormerPodRestartStatus whether there are any form pod is going to be restarted
+// return true if existed
+func checkFormerPodRestartStatus(kubeCli kubernetes.Interface, memberType v1alpha1.MemberType, tc *v1alpha1.TidbCluster, namespace string, ordinal int32, replicas int32) (bool, error) {
+	for i := replicas - 1; i > ordinal; i-- {
```

Comment:
ah, I'm wrong because we need to check the desired pods and return an error if the pod is not created yet, otherwise, pods can be deleted before the previous replica is recreated.

Reply:
I found that the admission pod webhook only considered the pod controlled by `apps.StatefulSet` for now. Though it would received the deleting request of `advancedStatefulset`, there are some places it still not support `AdvancedStatefulset` yet. https://github.com/pingcap/tidb-operator/blob/0bf625985d2b8ec35e845fffe31a135567695fd8/pkg/webhook/pod/util.go#L126
I think it should be supported in another issue to take it as one whole problem.

Reply:
the kind of `AdvancedStatefulset` is `StatefulSet` too.

Reply:
the only change here is to use `helper.GetPodOrdinals` to get the list of ordinals instead of `for loop`.

Reply:
the returned set has a method `List()` to get a sorted slice in asc order, all ordinals `<= ordinal` can be ignored in the loop

Reply:
I wonder if `advancedStatefulSet` is enabled, would the following method in `pkg/webhook/pod/util` get the owner advancedstatefulset ?
```golang
func getOwnerStatefulSetForTiDBComponent(pod *core.Pod, kubeCli kubernetes.Interface) (*apps.StatefulSet, error) {
	name := pod.Name
	namespace := pod.Namespace
	var ownerStatefulSetName string
	for _, ownerReference := range pod.OwnerReferences {
		if ownerReference.Kind == "StatefulSet" {
			ownerStatefulSetName = ownerReference.Name
			break
		}
	}
	if len(ownerStatefulSetName) == 0 {
		return nil, fmt.Errorf(failToFindTidbComponentOwnerStatefulset, namespace, name)
	}
	return kubeCli.AppsV1().StatefulSets(namespace).Get(ownerStatefulSetName, meta.GetOptions{})
}
```

Reply:
it will because it does not check the APIGroup (the only different between advanced statefulset with k8s statefulset is the APIGroup)
I'm fine to update later because the admission controller does not support `--features` flag yet.